### PR TITLE
chore(Space): modify space name and symbol

### PR DESCRIPTION
**Pull request by fedealconada**
## Motivation

Currently, Space pools do not have a meaningful name and symbol but they just take the same name as the adapter. 

## Solution
Following a similar logic to the one used for PTs and YTs, we want to name the Space pools this way.
E.g:
Name -> `1st May 2022 Compound USDC Space LP`
Symbol -> `Space-LP-cUSDC:01:05:2022`

We've added the `DateTime.sol` library which we took from Sense v1 but having to check it's pragma version to match the Space contracts (0.7.0).
We've created 2 internal methods that are called on the constructor which generate and return the name and symbol.

## Implementer Checklist
- [x]  [FIRST TIME ONLY] Review the [Solcurity](https://github.com/Rari-Capital/solcurity) standard
- [ ]  [OPTIONAL] Create a reference implementation (in python or JS)
- [x]  Create a short "Motivation" section for the PR
- [x]  Write a spec for the feature and put it in the PR description – basic function names and expected state transitions is OK
- [ ]  Check all of the new revert paths with concrete tests
- [ ]  Check all of the new storage slot writes with concrete tests – do not make the passing test case the trivial case!
- [x]  Go line-by-line through the new code and ensure it has all been covered by concrete tests – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
- [x]  Simplify the implementation and spend some time trying to minimize gas costs
- [ ]  Write fuzz tests for the feature – try everything to break the implementation (is this function monotonically in/decreasing, should it always be less than something else, etc)
- [ ]  Capture bugs discovered during the above steps in concrete tests (regression tests)
- [ ]  Add integration tests – fuzz or concrete tests that test how the new code affects the entire system, either locally or with a mainnet fork
- [ ]  [OPTIONAL] Integrate the feature into the deployment scripts
- [ ]  [OPTIONAL] Add new sanity checks to the deployment scripts
- [ ]  Get the PR reviewed by at least two people

### CODE REVIEW

File: src/utils/DateTime.sol [hunk 1]
```diff
@@ -0,0 +1,217 @@
+pragma solidity ^0.7.0;
+
+/// @author Taken from: https://github.com/bokkypoobah/BokkyPooBahsDateTimeLibrary
+library DateTime {
+    uint256 constant SECONDS_PER_DAY = 24 * 60 * 60;
+    uint256 constant SECONDS_PER_HOUR = 60 * 60;
+    uint256 constant SECONDS_PER_MINUTE = 60;
```
Comment (jparklev) [inline src/utils/DateTime.sol:7]: Mind if we satisfy these guys with explicit visability?
Screen Shot 2022-04-21 at 11 50 36 AM

File: src/tests/Space.t.sol [hunk 2]
```diff
@@ -146,6 +146,29 @@ contract SpaceTest is Test {
...
+            true
+        );
+        uint256 pti = pt < address(target) ? 0 : 1;
+        assertEq(pool.adapter(), address(adapter));
+        assertEq(pool.maturity(), maturity);
+        assertEq(pool.pti(), pti);
+        assertEq(pool.ts(), ts);
+        assertEq(pool.g1(), g1);
+        assertEq(pool.g2(), g2);
+        assertEq(pool.name(), "3rd July 1970 Target Token Space LP");
```
Comment (jparklev) [inline src/tests/Space.t.sol:168]: Maybe we should add a small test to check the datetime library with maturity closer today as well, one where we've verified the date externally so we can compare the string with our expectations. WDYT?

**Review by jparklev — APPROVED** [review]
We downgraded the datetime library here from `sense-v1` – is there any reason to think there's something in it that's incompatible with `0.7.0`?

- [x]  [FIRST TIME ONLY] Review the [[solcurity](https://github.com/Rari-Capital/solcurity)](https://github.com/Rari-Capital/solcurity) standard
- [x]  Read the PR descriptions and the linear ticket
- [x]  Do a high-level review of the code to confirm that the implementation does what it’s supposed to do
- [x]  Go line-by-line through the implementation and check state transitions – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
- [x]  Look at the feature and take some time to think about what tests *you* would write, then confirm that all of those cases are covered by the tests that are present
- [ ]  Confirm that the implementer has completed the implementer checklist correctly

**Comment by fedealconada** [comment]
> We downgraded the datetime library here from `sense-v1` – is there any reason to think there's something in it that's incompatible with `0.7.0`?
> 
> * [x]   [FIRST TIME ONLY] Review the [[solcurity](https://github.com/Rari-Capital/solcurity)](https://github.com/Rari-Capital/solcurity) standard
> * [x]   Read the PR descriptions and the linear ticket
> * [x]   Do a high-level review of the code to confirm that the implementation does what it’s supposed to do
> * [x]   Go line-by-line through the implementation and check state transitions – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
> * [x]   Look at the feature and take some time to think about what tests _you_ would write, then confirm that all of those cases are covered by the tests that are present
> * [x]   Confirm that the implementer has completed the implementer checklist correctly

I don't think so!

File: src/tests/Space.t.sol (hunk 2 above)
Comment (fedealconada) [reply src/tests/Space.t.sol:168]: done!

**Comment by jparklev** [comment]
Cool, if you've double checked on the sol version changes, we're good to merge!

**Pull request merged**